File: app/src/main/java/exceptionapi/ex02.java
package exceptionapi;

public class ex02 {
    static class InvalidAgeException extends RuntimeException {
        InvalidAgeException(String message) {
            super(message);
        }
    }

    static void checkAge(int age) {
        if (age < 0 || age > 150) {
            throw new InvalidAgeException("invalid age: " + age);
        }
    }

    public static void main(String[] args) {
        try {
            checkAge(-5);
        } catch (InvalidAgeException e) {
            System.out.println(e.getMessage());
        }
    }
}

File: app-kotlin/src/main/kotlin/exceptionapi/ex02.kt
package exceptionapi

class ex02 {
    class InvalidAgeException(message: String) : RuntimeException(message)

    companion object {
        fun checkAge(age: Int) {
            if (age < 0 || age > 150) {
                throw InvalidAgeException("invalid age: $age")
            }
        }

        @JvmStatic
        fun main(args: Array<String>) {
            try {
                checkAge(-5)
            } catch (e: InvalidAgeException) {
                println(e.message)
            }
        }
    }
}
